ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousHikingStateMachine/ReadyToPlanState.java

```diff
@@ -0,0 +1,102 @@
+package us.ihmc.behaviors.activeMapping.ContinuousHikingStateMachine;
+
+import behavior_msgs.msg.dds.ContinuousWalkingCommandMessage;
+import controller_msgs.msg.dds.FootstepDataListMessage;
+import us.ihmc.behaviors.activeMapping.ContinuousHikingParameters;
+import us.ihmc.behaviors.activeMapping.ContinuousPlanner;
+import us.ihmc.behaviors.activeMapping.ContinuousPlannerStatistics;
+import us.ihmc.behaviors.activeMapping.ControllerFootstepQueueMonitor;
+import us.ihmc.footstepPlanning.FootstepDataMessageConverter;
+import us.ihmc.footstepPlanning.monteCarloPlanning.TerrainPlanningDebugger;
+import us.ihmc.log.LogTools;
+import us.ihmc.perception.heightMap.TerrainMapData;
+import us.ihmc.robotics.stateMachine.core.State;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ReadyToPlanState implements State
+{
+   private final AtomicReference<ContinuousWalkingCommandMessage> commandMessage;
+   private final ContinuousPlanner continuousPlanner;
+   private final ControllerFootstepQueueMonitor controllerFootstepQueueMonitor;
+   private final ContinuousHikingParameters continuousHikingParameters;
+   private final TerrainMapData terrainMap;
+   private final TerrainPlanningDebugger debugger;
+   private final ContinuousPlannerStatistics statistics;
+   private final StepValidityChecker stepValidityChecker;
+
+   public ReadyToPlanState(AtomicReference<ContinuousWalkingCommandMessage> commandMessage,
+                           ContinuousPlanner continuousPlanner,
+                           ControllerFootstepQueueMonitor controllerFootstepQueueMonitor,
+                           ContinuousHikingParameters continuousHikingParameters,
+                           TerrainMapData terrainMap,
+                           TerrainPlanningDebugger debugger,
+                           ContinuousPlannerStatistics statistics,
+                           StepValidityChecker stepValidityChecker)
+   {
+      this.commandMessage = commandMessage;
+      this.continuousPlanner = continuousPlanner;
+      this.controllerFootstepQueueMonitor = controllerFootstepQueueMonitor;
+      this.continuousHikingParameters = continuousHikingParameters;
+      this.terrainMap = terrainMap;
+      this.debugger = debugger;
+      this.statistics = statistics;
+      this.stepValidityChecker = stepValidityChecker;
+   }
+
+   @Override
+   public void onEntry()
+   {
+      LogTools.warn(String.format("Entering %s", getClass().getSimpleName()));
+
+      continuousPlanner.initialize();
```

Comment:
We should assume whenever we enter this state, that nothing has been initialized, and that we have no plan, because in the doAction() we will play and the isDone() condition will be set to true once we get a plan